### PR TITLE
feat(build_config): extras_require

### PR DESCRIPTION
**Pull request by aarnphm**
update logs instruction when running inside docker container

specifying build options via `python`:
```yaml
python:
  extras_require:
    - tracing
    - grpc
```

### CODE REVIEW

File: bentoml/_internal/bento/build_config.py [hunk 1]
```diff
@@ -64,9 +64,7 @@ def _convert_python_version(py_version: str | None) -> str | None:
     target_python_version = f"{major}.{minor}"
     if target_python_version != py_version:
         logger.warning(
-            "BentoML will install the latest python%s instead of the specified version %s. To use the exact python version, use a custom docker base image. See https://docs.bentoml.org/en/latest/concepts/bento.html#custom-base-image-advanced",
-            target_python_version,
-            py_version,
+            f"BentoML will install the latest 'python{target_python_version}' instead of the specified 'python{py_version}'. To use the exact python version, use a custom docker base image. See https://docs.bentoml.org/en/latest/concepts/bento.html#custom-base-image-advanced",
```
Comment (parano) [inline bentoml/_internal/bento/build_config.py:67]: f string will be slower than the original code

Comment (aarnphm) [reply bentoml/_internal/bento/build_config.py:67]: really?

Comment (aarnphm) [reply bentoml/_internal/bento/build_config.py:67]: https://docs.python.org/3/howto/logging.html#optimization

Comment (parano) [reply bentoml/_internal/bento/build_config.py:67]: yup

**Pull request merged**